crud.py

```diff
@@ -0,0 +1,48 @@
+from datetime import date
+
+from sqlalchemy.orm import Session
+
+import models, schemas
+
+
+def get_all_authors(db_session: Session):
+    authors = db_session.query(models.Author).all()
+    return authors
+
+def get_author_by_id(db_session: Session, author_id: int):
+    return db_session.query(models.Author).filter(models.Author.id == author_id).first()
+
+
+def create_author(db_session: Session, author: schemas.AuthorCreate):
+    db_new_author = models.Author(name=author.name, bio=author.bio)
+    db_session.add(db_new_author)
+    db_session.commit()
+    db_session.refresh(db_new_author)
+    return db_new_author
+
+
+def get_all_books(db_session: Session, author_id: int | None = None):
```

Comment:
The use of `int | None` for type hinting requires Python 3.10 or later. If you need to support earlier versions of Python, consider using `Optional[int]` from the `typing` module instead.